test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectProtocolValidatorJsonWebTokenTests.cs

```diff
@@ -0,0 +1,1700 @@
+                {
+                    PropertyNamesAndSetGetValue = new List<KeyValuePair<string, List<object>>>
+                    {
+                        new KeyValuePair<string, List<object>>("NonceLifetime", new List<object>{TimeSpan.FromMinutes(60), TimeSpan.FromMinutes(10), TimeSpan.FromMinutes(100)}),
+                        new KeyValuePair<string, List<object>>("RequireAcr", new List<object>{false, true, false}),
+                        new KeyValuePair<string, List<object>>("RequireAmr", new List<object>{false, true, false}),
+                        new KeyValuePair<string, List<object>>("RequireAuthTime", new List<object>{false, true, false}),
+                        new KeyValuePair<string, List<object>>("RequireAzp", new List<object>{false, true, false}),
+                        new KeyValuePair<string, List<object>>("RequireNonce", new List<object>{true, false, true}),
+                        new KeyValuePair<string, List<object>>("RequireSub", new List<object>{true, false, true}),
+                        new KeyValuePair<string, List<object>>("RequireTimeStampInNonce", new List<object>{true, false, true}),
+                        new KeyValuePair<string, List<object>>("RequireState", new List<object>{true, false, true}),
+                        new KeyValuePair<string, List<object>>("RequireStateValidation", new List<object>{true, false, true}),
+                    },
+                    Object = validationParameters,
+                };
+
+            TestUtilities.GetSet(context);
+            TestUtilities.AssertFailIfErrors("OpenIdConnectProtocolValidator_GetSets", context.Errors);
+
+            ExpectedException ee = ExpectedException.ArgumentNullException();
+            Assert.NotNull(validationParameters.HashAlgorithmMap);
+            Assert.Equal(18, validationParameters.HashAlgorithmMap.Count);
+
+            ee = ExpectedException.ArgumentOutOfRangeException();
+            try
+            {
+                validationParameters.NonceLifetime = TimeSpan.Zero;
+                ee.ProcessNoException();
+            }
+            catch (Exception ex)
+            {
+                ee.ProcessException(ex);
+            }
+
+            ee = ExpectedException.ArgumentNullException();
+            try
+            {
+                validationParameters.CryptoProviderFactory = null;
+                ee.ProcessNoException();
+            }
+            catch (Exception ex)
+            {
+                ee.ProcessException(ex);
+            }
+        }
+
+
+        [Theory, MemberData(nameof(ValidateAuthenticationResponseTheoryData))]
+        public void ValidateAuthenticationResponse(OidcProtocolValidatorJsonWebTokenTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ValidateAuthenticationResponse", theoryData);
+            try
+            {
+                theoryData.ProtocolValidator.ValidateAuthenticationResponse(theoryData.ValidationContext);
```

Comment:
I don't think that we have a test for ValidateAuthenticationResponse method which will test the whole method, including Validate State/IdToken/Nonce.. method calls.
It looks like we didn't have such a test earlier, but it would be valuable to have it, imo.

Reply:
I added two tests (one using JwtSecurityToken and the other using JsonWebToken) that test this entire method.